### PR TITLE
Api provides admin functionality to create article

### PR DESCRIPTION
**Pull request by LucasKnudsen**
[**PT-Story:** ](https://www.pivotaltracker.com/story/show/178092075) 
### User Story 
``` 
As an Api
In order for an journalist to create an article
I need to be able to handle a submitted form and create an article based on this data 
``` 
## Changes proposed in this pull request: 
* Adds create action for Article controller
* Limits article creation to 'journalist' role
* Adds error handlers for unauthenticated user, for unauthorized user, and for lacking params.
## What I have learned working on this feature: 
* Mass assignment with .permit()
* That render json: doesn't halt execution, but we can use `and return` for this instead
* render json: in private methods will still continue on over to the main action method
* Role based conditionals

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -17,9 +19,21 @@ def index
...
   def show
     article = Article.find(params[:id])
-
     render json: article, serializer: ArticlesShowSerializer
   rescue ActiveRecord::RecordNotFound => e
     render json: { error_message: "Article does not exist" }, status: 404
   end
+
+  def create
+    unless current_user.role == "journalist"
```
Comment (tochman) [inline app/controllers/api/articles_controller.rb]: If you are using `enum` you can simplify this

```ruby
current_user.journalist?
```

Comment (LucasKnudsen) [reply app/controllers/api/articles_controller.rb]: Good point!

File: app/controllers/api/articles_controller.rb [hunk 2]
```diff
@@ -17,9 +19,21 @@ def index
...
+      render json: { error_message: "You dont have access" }, status: 403 and return
+    end
+
+    article = current_user.articles.create(params[:article].permit(:title, :teaser, :body, :category))
```
Comment (tochman) [inline app/controllers/api/articles_controller.rb]: Not a blocker, but a good practice is to extract this to a `private` method

```ruby
def article_params
   params[:article].permit(:title, :teaser, :body, :category)
end
```

and then use it:

```ruby
article = current_user.articles.create(article_params)
```

File: spec/requests/api/auth/journalist_can_create_articles_spec.rb [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  describe "successfully" do
+    before do
+      post "/api/articles",
+           params: {
+             article: {
+               title: "Obnoxious Title",
+               teaser: "Some damn teaser",
+               body: "Husband found dead allegedly because he wasn't testing first",
+               category: "Hollywood",
+               user_id: journalist.id,
```
Comment (tochman) [inline spec/requests/api/auth/journalist_can_create_articles_spec.rb]: There is no need to pass in `user_id`, you are authenticating with a token and this attribute is not used in your implementation.

File: app/controllers/api/articles_controller.rb (hunk 2 above)
Comment (LucasKnudsen) [reply app/controllers/api/articles_controller.rb]: Good point x2!

File: spec/requests/api/auth/journalist_can_create_articles_spec.rb [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+           },
+           headers: auth_headers_member
+    end
+
+    it "is expected to have status 403" do
+      expect(response).to have_http_status 403
+    end
+
+    it "is expected that member cant create article" do
+      expect(response_json["error_message"]).to eq "You dont have access"
```
Comment (tochman) [inline spec/requests/api/auth/journalist_can_create_articles_spec.rb]: `dont`. -> `don't` 

But the whole message feels cryptic to me

File: spec/requests/api/auth/journalist_can_create_articles_spec.rb (hunk 3 above)
Comment (LucasKnudsen) [reply spec/requests/api/auth/journalist_can_create_articles_spec.rb]: Ye, not sure what's going on there actually :3

File: spec/requests/api/auth/journalist_can_create_articles_spec.rb [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    end
+
+    it "is expected not to add an article to the database" do
+      articles = Article.all
+      expect(articles.count).to eq 0
+    end
+  end
+
+  describe "unccessfully" do
```
Comment (lukeperrin10) [inline spec/requests/api/auth/journalist_can_create_articles_spec.rb]: Just a small typo here on line 65

File: spec/requests/api/auth/journalist_can_create_articles_spec.rb [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+    it "is expected to have status 401" do
+      expect(response).to have_http_status 401
+    end
+
+    it "is expected that journalist cant create article" do
+      expect(response_json["errors"].first).to eq "You need to sign in or sign up before continuing."
+    end
+  end
+  describe "unccessfully with missing params" do
```
Comment (lukeperrin10) [inline spec/requests/api/auth/journalist_can_create_articles_spec.rb]: Here aswell

Comment (LucasKnudsen) [reply spec/requests/api/auth/journalist_can_create_articles_spec.rb]: we unccessed :(

**Review by 1CIM — COMMENTED** [review]
looks like a little different solar graph settings.

**Review by tochman — APPROVED** [review]
Looks great 🤩

**Pull request merged**